File: frontend/src/services/inventory.ts
import api from './api';

export interface ProductInventoryItem {
  id: number;
  company_id: number;
  product_id: number;
  product_name: string;
  qty_available: number;
}

export interface ProductSale {
  id: number;
  company_id: number;
  product_id: number;
  product_name: string;
  quantity: number;
  unit_price: number;
  total_amount: number;
  customer_name: string | null;
  sale_date: string;
  notes: string | null;
  created_at: string;
}

export interface ProductSaleCreate {
  product_id: number;
  quantity: number;
  unit_price: number;
  customer_name?: string;
  sale_date: string;
  notes?: string;
}

export const inventoryService = {
  async getProductInventory(): Promise<ProductInventoryItem[]> {
    const response = await api.get('/inventory/products/');
    return response.data;
  },

  async getMaterialInventory(): Promise<any[]> {
    // Materials inventory is just materials with qty_available
    const response = await api.get('/materials/');
    return response.data;
  },

  async getProductSales(): Promise<ProductSale[]> {
    const response = await api.get('/inventory/products/sales');
    return response.data;
  },

  async registerSale(saleData: ProductSaleCreate): Promise<ProductSale> {
    const response = await api.post('/inventory/products/sales', saleData);
    return response.data;
  },
};

File: backend/app/models/inventory.py
from datetime import datetime, date
from sqlalchemy import Column, Integer, ForeignKey, DateTime, Numeric, String, Date, Text
from sqlalchemy.orm import relationship
from app.database import Base


class ProductInventory(Base):
    """Inventory of finished products"""
    __tablename__ = "product_inventory"
    
    id = Column(Integer, primary_key=True, index=True)
    company_id = Column(Integer, ForeignKey("companies.id"), nullable=False, index=True)
    product_id = Column(Integer, ForeignKey("products.id"), nullable=False, index=True)
    qty_available = Column(Numeric(10, 2), default=0, nullable=False)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # Relationships
    company = relationship("Company")
    product = relationship("Product")


class ProductSale(Base):
    """Sales of finished products from inventory"""
    __tablename__ = "product_sales"
    
    id = Column(Integer, primary_key=True, index=True)
    company_id = Column(Integer, ForeignKey("companies.id"), nullable=False, index=True)
    product_id = Column(Integer, ForeignKey("products.id"), nullable=False, index=True)
    quantity = Column(Numeric(10, 2), nullable=False)
    unit_price = Column(Numeric(10, 2), nullable=False)
    customer_name = Column(String, nullable=True)
    sale_date = Column(Date, nullable=False, default=date.today, index=True)
    notes = Column(Text, nullable=True)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    
    # Relationships
    company = relationship("Company")
    product = relationship("Product")
